libbeat/idxmgmt/idxmgmt.go

```diff
@@ -57,25 +51,43 @@ type Supporter interface {
 
 	// Manager creates a new manager that can be used to execute the required steps
 	// for initializing an index, ILM policies, and write aliases.
-	Manager(client ESClient, assets Asseter) Manager
+	Manager(client Client, assets Asseter) Manager
 }
 
 // Asseter provides access to beats assets required to load the template.
 type Asseter interface {
 	Fields(name string) []byte
 }
 
-// ESClient defines the minimal interface required for the index manager to
+// Client defines the minimal interface required for the index manager to
 // prepare an index.
-type ESClient interface {
+type Client interface {
 	Request(method, path string, pipeline string, params map[string]string, body interface{}) (int, []byte, error)
 	GetVersion() common.Version
 }
 
 // Manager is used to initialize indices, ILM policies, and aliases within the
 // Elastic Stack.
 type Manager interface {
-	Setup(forceTemplate, forcePolicy bool) error
+	Setup(template, ilm SetupConfig) error
+}
+
+// SetupConfig is used to define loading and forcing to overwrite during the setup process
+type SetupConfig struct {
+	Load  *bool
+	Force *bool
+}
```

Comment:
Why no enum or bin patterns, so we can give the valid states a name?

Load and Force can have 3 states each, getting is a total of 9. We have only 2 getters on this type, which makes SetupConfig itself look like it has only 3 valid states: NoLoad, LoadCreate, LoadForce.

The zero value of SetupConfig indicates that LoadForce is the default, whereas DefaultSetupConfig makes LoadCreate the default. 

We use pointers in config objects to be read from config file to indicate that a setting has not been set at all. But SetupConfig is internal only, right? Better not simulate a tri-state variable via pointers.

Do we always want `Force` to be true by default always?

Reply:
The `nil` check is actually used [here](https://github.com/elastic/beats/blob/f5e64b51032584dfaaf7f6f3d3adacc506a38d72/libbeat/idxmgmt/std.go#L191). There might be a better way of doing that, I didn't dive too deep into the `m.support.st.withILM.Load()` handling.